cartographer_ros/cartographer_ros/node.cc

```diff
@@ -220,6 +228,13 @@ void Node::PublishTrajectoryStates(const ::ros::WallTimerEvent& timer_event) {
   }
 }
 
+void Node::PublishTrajectoryNodesList(
+    const ::ros::WallTimerEvent& unused_timer_event) {
+  carto::common::MutexLocker lock(&mutex_);
+  trajectory_nodes_list_publisher_.publish(
```

Comment:
Maybe a check for subscriber count > 0 could be added here, in order to avoid running GetTrajectoryNodesList for nothing.